src/Data/LLVM/BitCode/IR/Metadata.hs

```diff
@@ -611,8 +611,16 @@ parseMetadataEntry vt mt pm (fromEntry -> Just r) =
         <*> parseField r 10 numeric                                 -- didtFlags
         <*> (mdForwardRefOrNull ctx mt <$> parseField r 11 numeric) -- didtExtraData
         <*> (if length (recordFields r) <= 12
-             then pure Nothing
-             else Just                 <$> parseField r 12 numeric) -- didtDwarfAddressSpace
+             then pure Nothing  -- field not present
+             else do v <- parseField r 12 numeric
+                     -- dwarf address space is encoded in bitcode as +1; a value
+                     -- of zero means there is no dwarf address space present.
+                     -- The AST representation is the actual address space, or
+                     -- Nothing if there is no address space (indistinguishable
+                     -- from "field not present" for LLVM 4 and earlier).
```

Comment:
Worth citing the place in the LLVM source code where this convention is described.